src/toolbox/shamir_secrets.rs

```diff
@@ -112,40 +239,24 @@ impl SecretShare {
             sum += delta;
         }
 
+        let mut xis: Vec<Scalar> = Vec::new();
+        for i in 1..(threshold+1) {
```

Comment:
Guessing this, but something like:
`xis = (1..(threshold+1)).map(|i| Scalar::from(i as u64)).collect()`

Reply:
THAT'S how you do it.  I was trying to figure out a `map()` invocation that would handle generating `xis` but couldn't find it.  This is pretty simple, so I'm not sure how I missed it.  Thanks!